src/object/cli_obj.c

```diff
@@ -3392,6 +3392,15 @@ obj_shard_list_comp_cb(struct shard_auxi_args *shard_auxi,
 
 	shard_arg = container_of(shard_auxi, struct shard_list_args, la_auxi);
 	if (obj_auxi->req_tgts.ort_grp_size == 1) {
+		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE) {
+			int i;
+
+			for (i = 0; i < shard_arg->la_nr; i++) {
+				if (shard_arg->la_recxs[i].rx_idx & PARITY_INDICATOR)
+					obj_recx_parity_to_daos(obj_get_oca(obj_auxi->obj),
+								&shard_arg->la_recxs[i]);
```

Comment:
just confirm, for the case of full-stripe update with partial overwrite, for example EC4+1 cell size 1M.
firstly write [0-4M]@epoch10, and overwrite [0-1M]@epoch20.
If enumerate from parity shard, it will get back two recx [0-1M] and [0, P | 1M], for this case it should finally report only one recx which is [0-4M] back to user right?
but did not see where do that kind of merge? there is some merge handling in obj_ec_recxs_convert(), but seems for parity ext it do obj_recx_parity_to_daos and without merge with possibly overlapped replica exts?

Reply:
thx. Not a problem for this patch, but in obj_ec_recxs_convert(), 
3160 >       if (recx->rx_idx & PARITY_INDICATOR) {                                                      
3161 >       >       obj_recx_parity_to_daos(oca, recx);                                                 
3162 >       >       return 0;                                                                           
3163 >       }     
there seems also need a merge, if the parity ext is the last recx but if have overlap with previous replica recx?